crates/dojo-lang/src/compiler.rs

```diff
@@ -400,54 +416,67 @@ fn get_dojo_contract_artifacts(
                 .get(&module_name as &str)
                 .map_or_else(Vec::new, |write_ops| find_module_rw(db, module_id, write_ops));
 
-            let contract = Contract {
-                name: module_name.clone(),
-                class_hash: *class_hash,
-                abi: abi.clone(),
-                writes,
-                reads,
-                ..Default::default()
-            };
-
-            result.insert(module_name, contract);
+            let manifest = Manifest::new(
+                DojoContract {
+                    writes,
+                    reads,
+                    class_hash: *class_hash,
+                    abi: None,
+                    ..Default::default()
+                },
+                module_name.clone(),
+            );
+
+            result.insert(module_name, (manifest, abi.clone()));
         }
     }
 
     Ok(result)
 }
 
-fn do_update_manifest(
-    current_manifest: &mut dojo_world::manifest::Manifest,
-    world: dojo_world::manifest::Contract,
-    base: dojo_world::manifest::Class,
-    resource_metadata: dojo_world::manifest::Contract,
-    models: BTreeMap<String, dojo_world::manifest::Model>,
-    contracts: BTreeMap<SmolStr, dojo_world::manifest::Contract>,
-) -> anyhow::Result<()> {
-    if current_manifest.world.class_hash != world.class_hash {
-        current_manifest.world = world;
+fn write_manifest_and_abi<T>(
+    relative_manifest_dir: &Utf8PathBuf,
+    relative_abis_dir: &Utf8PathBuf,
+    manifest_dir: &Utf8PathBuf,
+    manifest: &mut Manifest<T>,
+    abi: &Option<abi::Contract>,
+) -> anyhow::Result<()>
+where
+    T: Serialize + DeserializeOwned + ManifestMethods,
+{
+    let parts: Vec<&str> = manifest.name.split("::").collect();
+    let name: Utf8PathBuf = parts.last().unwrap().into();
```

Comment:
We may want to take only `manifest.name` if this unwrap fails?

Reply:
[split](https://doc.rust-lang.org/std/primitive.str.html#method.split) always return iterator with atleast 1 element so pattern in `split` doesn't match we will still have `parts[0]` as full `manifest.name`